### DISCUSSION TITLE
特定キー反応無し、はんだ盛り、ダイオード確認済

### DISCUSSION THREAD

**Discussion by Yuumarusan918** [Yowkees/keyball#520] [Q&A]
Dキーが反応しません。何度か症状が再発しており、その度にはんだを盛ると解決しましたが、今回は解決する兆しがありません。
前はキーを強く押すと認識されましたが、今回は認識されず

ピンで導通確認後は認識されるようになる、といった謎が多い不具合が発生しています。どうしたら治るでしょうか？考えうる原因か改善策がありましたらご教示頂きたいです。

**Comment by koron** [comment]
(issueからdiscussionへ変換しました)

Dキーだけということであれば、Dキーに使ってるキースイッチを別のものに交換してみてください。

回路や基板やPro Microに問題があれば、通常ならばDの上下左右、周りのキーも反応しない可能性が高いです。
そうでないということはキースイッチの疑いが一番強くなります。

次に疑うのはDキーのキースイッチソケットの電極が緩くなってるケースです。
キースイッチを抜いた状態で他のキーのソケットと電極部分の隙間を見比べてみてください。
もしも広くなっているようでしたら、先のとがったもので電極を狭めるなどの対策が打てます。
もしくはキースイッチソケットを交換しても良いでしょう。

あとはDキーに対応するダイオードが破損している可能性も考えられます。
その場合はダイオードを交換すると良いでしょう。
どのダイオードがDキーに対応するものかは、
Keyballの種類や左右どちらにボールを置いて組み立てたかで変わります。
